### ISSUE TITLE
@SuppressWarnings("all") on generated Java classes is ineffective

### ISSUE THREAD

**Issue by trejkaz** [antlr/antlr3#186]
javac apparently ignores `@SuppressWarnings("all")`. Therefore, this should be replaced by the list of actual warnings ANTLR's generated code triggers, to actually suppress the warnings.

I'm not exactly aware of the complete set of warnings it can create, though.

**Comment by timeraider4u** [comment]
Which java compiler are you using that is ignoring the "all" statement? Mine (oracle-jdk 1.8.0.121) is quiet fine with it:
```bash
$ echo -e '@SuppressWarnings("all")\npublic class Hello {\n\tpublic static void main (String [] args) {\n\t\tint a=0;\n\t}\n}'  > /tmp/Hello.java && javac Hello.java && echo $?
0
```

**Comment by trejkaz** [comment]
1.8.0_92 here, but that particular example prints 0 with or without the SuppressWarnings line.

**Comment by trejkaz** [comment]
Also tried on Java 9-ea, since that happened to be sitting around on my computer, and same result on that version.

**Comment by timeraider4u** [comment]
Yes, with the `-Xlint:all` command line option its now the same for me. Seems I am not using it anywhere...

**Comment by trejkaz** [comment]
This makes some sense then. I guess "cast" type warnings are disabled by default.

**Comment by timeraider4u** [comment]
Ah okay, according to http://docs.oracle.com/javase/7/docs/technotes/tools/windows/javac.html#nonstandard right, `-Xlint` with some optional argument is enabling the according warnings.

Interestingly the following is working:
```bash
~ $ echo -e '@SuppressWarnings("cast")\npublic class Hello {\n\tpublic static void main (String [] args) {\n\t\tString a = (String) "a";\n\t}\n}' > /tmp/Hello.java && cat /tmp/Hello.java && javac -Xlint:all -Werror /tmp/Hello.java && echo $?
@SuppressWarnings("cast")
public class Hello {
        public static void main (String [] args) {
                String a = (String) "a";
        }
}
0
```
while this example fails:
```bash
echo -e '@SuppressWarnings("all")\npublic class Hello {\n\tpublic static void main (String [] args) {\n\t\tString a = (String) "a";\n\t}\n}' > /tmp/Hello.java && cat /tmp/Hello.java && javac -Xlint:all -Werror /tmp/Hello.java && echo $?
@SuppressWarnings("all")
public class Hello {
        public static void main (String [] args) {
                String a = (String) "a";
        }
}
/tmp/Hello.java:4: warning: [cast] redundant cast to String
                String a = (String) "a";
                           ^
error: warnings found and -Werror specified
1 error
1 warning
```

I would consider this a bug in the javac-compiler itself but maybe I am missing something

**Comment by trejkaz** [comment]
I think it's just that "all" is not a class of warning, and not handled specially by the compiler either, so it doesn't suppress anything.

The confusing thing is that IDEs do treat it as a special value.

**Comment by timeraider4u** [comment]
According to the Java specification at http://docs.oracle.com/javase/specs/jls/se7/html/jls-9.html#jls-9.6.3.5 its up to Java implementations if "all" is a valid value.

E.g., for Eclipse the suppressing string "all" is well defined - see http://help.eclipse.org/kepler/index.jsp?topic=%2Forg.eclipse.jdt.doc.user%2Ftasks%2Ftask-suppress_warnings.htm

**Comment by trejkaz** [comment]
Yeah. I also remember that previous versions of either Eclipse or IDEA didn't recognise "all", but _did_ recognise "ALL", so in some places in our code we ended up with `@SuppressWarnings({"all", "ALL"})`.

So anyway, it seems that we'd want "all" and "cast", even though it seems counterintuitive...

Or I guess the warnings could be fixed. In my case, at least, every single cast warning is about a cast to Object.
